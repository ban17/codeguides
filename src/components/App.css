.code-guides-app {
  font-size: 2rem;
  font-family: "Roboto Light", sans-serif;
  text-align: center;

  header {
    padding: 1rem 0;
    color: darkblue;
    background-color: bisque;

    > h1 {
      margin: 0;
      font-size: 3.5rem;
    }
  }

  .content {
    padding: 2rem 0;
  }

  footer {
    position: absolute;
    bottom: 0;
    width: 100%;
    padding: 1rem 0;
    color: white;
    background-color: darkcyan;

    > h2 {
      margin: 0;
      font-size: 3rem;
      letter-spacing: 1rem;
    }
  }

  button {
    display: block;
    width: 30rem;
    margin: 2rem auto;
    padding: 0.5rem 0;
    color: yellow;
    letter-spacing: 0.2rem;
    background-color: darkslateblue;
    border: none;
    border-radius: 1rem;
    outline: none;

    &:hover {
      color: white;
      background-color: darkcyan;
    }

    > a {
      color: yellow;
      text-decoration: none;
    }
  }
}
