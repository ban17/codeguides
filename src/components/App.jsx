import React, { useState } from 'react'

const App = () => {
    const [isOpen, changeIsOpen] = useState(false)

    return (
        <div className='app'>
            <h1>Code Guides App</h1>
            <button onClick={() => changeIsOpen(!isOpen)}>{ isOpen ? 'Hide current year' : 'Show current year' }</button>
            {isOpen && <h2>{new Date().getFullYear()}</h2>}
        </div>
    )
}

export default App
