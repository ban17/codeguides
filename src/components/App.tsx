import React, { useState } from 'react'

const App: React.FC = () => {
  const [isOpen, changeIsOpen] = useState<boolean>(false)

  return (
    <div className="code-guides-app">
      <h1>Code Guides App</h1>
      <h4>some text</h4>
      <button onClick={() => changeIsOpen(!isOpen)}>
        {isOpen ? 'Hide current year' : 'Show current year'}
      </button>
      {isOpen && <h2>{new Date().getFullYear()}</h2>}
    </div>
  )
}

export default App
