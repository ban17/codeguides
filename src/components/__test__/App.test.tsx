import { mount, shallow } from 'enzyme'
import React from 'react'

import App from '@/components/App'

describe('App', () => {
  it('should render App component', () => {
    expect(shallow(<App />)).toMatchSnapshot()
  })

  it('should show current year on button click', () => {
    const wrapper = mount(<App />)
    const button = wrapper.find('.test-button')

    button.simulate('click')
    wrapper.update()

    expect(wrapper.find('h3').text()).toBe(new Date().getFullYear().toString())
    wrapper.unmount()
  })
})
