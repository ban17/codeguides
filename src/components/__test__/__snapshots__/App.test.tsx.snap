// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component 1`] = `
<div
  className="code-guides-app"
>
  <h1>
    Code Guides App
  </h1>
  <h4>
    some text
  </h4>
  <button
    onClick={[Function]}
  >
    Show current year
  </button>
</div>
`;
